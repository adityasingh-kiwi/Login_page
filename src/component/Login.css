.login_page{
    position: relative;
    
    background-image: url("/home/kiwitech/Desktop/Project/login_page/src/image/login.jpg");
    background-size: cover;
    overflow-y: hidden;
}
