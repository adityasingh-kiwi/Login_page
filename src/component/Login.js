import React, { useState } from 'react'
import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import './Login.css';
import { useNavigate } from "react-router-dom";

const Login = (props) => {
  const [data, setData] = useState({
    userName: "",
    userEmail: "",
    userPassword: ""

  })

  let Naviagte = useNavigate();

  const handleChange = (e) => {
    console.log("e------------->>>", e.target.value, e.target.name);
    setData({ ...data, [e.target.name]: e.target.value })

  }


  // const history = useHistory();

  const handleLogin = async () => {
    console.log(props);

    const { userName, userEmail, userPassword } = data;
    const rest = await fetch("https://login-page-9c380-default-rtdb.asia-southeast1.firebasedatabase.app/loginpage.json", {
      method: "POST",
      headers: {
        "Content-type": "application/json",

      },
      body: JSON.stringify({
        userName,
        userEmail,
        userPassword

      })
    })

    if (rest.status == 200) {
      Naviagte("/home");

    }
    else {
      <div>
        <h1>page not found</h1>
      </div>
    }



  }
  //vh-100 gradient-custom
  return (
    <div className="login_page">


      <section className="">

        <div classNameName="container py-5 h-100">
          <div classNameName="row d-flex justify-content-center align-items-center h-100">
            <div className="col-12 col-md-8 col-lg-6 col-xl-5">
              <div className="card bg-dark text-white" style={{ "bordeRadius": "1rem" }}>
                <div className="card-body p-5 text-center">

                  <div className="mb-md-5 mt-md-4 pb-5">

                    <h2 className="fw-bold mb-2 text-uppercase">Login</h2>
                    <p className="text-white-50 mb-5">Please enter your login and password!</p>



                    <div className="form-outline form-white mb-4">
                      <input name="userName" type="text" id="typeEmailX" className="form-control form-control-lg" value={data.userName} onChange={handleChange} placeholder="username" required />
                      <label className="form-label" for="userName" >Name</label>
                    </div>
                    <div className="form-outline form-white mb-4">
                      <input name="userEmail" type="email" id="typeEmailX" className="form-control form-control-lg" value={data.userEmail} onChange={handleChange} placeholder="email id" required />
                      <label className="form-label" for="userEmail" >Email</label>
                    </div>

                    <div className="form-outline form-white mb-4">
                      <input name="userPassword" type="password" id="typePasswordX" className="form-control form-control-lg" value={data.userPassword} onChange={handleChange} placeholder="password" required />
                      <label className="form-label" for="userPassword">Password</label>
                    </div>

                    <p className="small mb-5 pb-lg-2"><a className="text-white-50" href="#!">Forgot password?</a></p>

                    <button className="btn btn-outline-light btn-lg px-5" type="submit" onClick={handleLogin}>Login</button>

                    {/* <div className="d-flex justify-content-center text-center mt-4 pt-1">
                      <a href="#!" className="text-white"><i className="fab fa-facebook-f fa-lg"></i></a>
                      <a href="#!" className="text-white"><i className="fab fa-twitter fa-lg mx-4 px-2"></i></a>
                      <a href="#!" className="text-white"><i className="fab fa-google fa-lg"></i></a>
                    </div> */}

                  </div>

                  {/* <div>
                    <p className="mb-0">Don't have an account? <a href="#!" className="text-white-50 fw-bold">Sign Up</a></p>
                  </div> */}

                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  )
}

export default Login
