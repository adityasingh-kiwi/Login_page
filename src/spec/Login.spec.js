import Login from '../component/Login';
import { mount, render, shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ExceptionMap } from 'antd/lib/result';

configure({ adapter: new Adapter() });

describe('Login_page',()=>{
   it('object check',()=>{
        let wrapper= shallow(<Login/>);
        expect(wrapper.exists('.login_page')).toEqual(true);
           
   
   })

 it('Login_Button_check',()=>{
     let wrapper= shallow(<Login/>);
     const container = wrapper.find(".login_button")
       expect(container.length).toEqual(1)
     wrapper.find('.login_button').simulate('click');


 })
 it("Check_handle_change_userEmail", () => {
    const wrapper = mount(<Login />);
    expect(wrapper.exists('#typeEmailX')).toEqual(true);
    wrapper.find('#typeEmailX').simulate('change', {
      target: {
        name:"userEmail",
        value: 'adityasingh@gmail.com',
      },
    }); 
    expect(wrapper.find('input[type="email"]').prop('value')).toEqual(
      'adityasingh@gmail.com',
    );
    
 
  });
   it("Check_handle_change_userEmail", () => {
     const wrapper = shallow(<Login />);
     const input = wrapper.find(".login_userEmail");
     input.simulate("change", { target: { name:"userEmail",value: " " } }); 
     expect(wrapper.find('input[type="email"]').prop('value')).toEqual(
      '',
    );
    
 
   });
    
  it("Check_handle_change_userPassword", () => {
    const wrapper = shallow(<Login />);
    const input = wrapper.find(".login_userPassword");
    input.simulate("change", { target: { name: 'userPassword',value: " " } });

  
  });


  it("Check_handle_change_userPassword", () => {
    const wrapper = shallow(<Login />);
    const input = wrapper.find(".login_userPassword");
    input.simulate("change", { target: { value: "1234567" } }); 
  
  });
    it('Snapshot_login',()=>{
      let wrapper=shallow(<Login/>)
      expect(wrapper).toMatchSnapshot();


    })
    it('should have the login false by default', () => {
      const container = mount(<Login />);
      expect(
        container.find('.login_button').prop('false'),
      ).toBeFalsy();
    });
     it('checking Login flow',()=>{
       let wrapper=shallow(<Login/>)
       const btn=wrapper.find('.login_button')
       wrapper.find('.login_button').simulate('click');
       expect(
        btn.find('.login_button').prop('false'),
        ).toBeFalsy();

     })



     it('handle-Login flow',()=>{
      let wrapper=shallow(<Login/>)
      expect(wrapper.exists('#typeEmailX')).toEqual(true);
      wrapper.find('#typeEmailX').simulate('change', {
      target: {
        name:"userEmail",
        value: 'adityasingh@gmail.com',
      },
    }); 

    const input = wrapper.find(".login_userPassword");
    input.simulate("change", { target: { value: "1234567" } }); 

    const btn = wrapper.find('.login_button')
     wrapper.find('.login_button').simulate('click');
     expect(
      wrapper.find('#error_paragraph').prop('defaultValue'),
      ).toBeFalsy();
      
    })

      



 })
 

