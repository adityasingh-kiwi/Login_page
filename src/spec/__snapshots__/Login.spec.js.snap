// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login_page Snapshot_login 1`] = `
<div
  className="login_page"
>
  <section
    className=""
  >
    <div
      className="container py-5 h-100"
    >
      <div
        className="row d-flex justify-content-center align-items-center h-100"
      >
        <div
          className="col-12 col-md-5 col-lg-6 col-xl-5"
        >
          <div
            className="card bg-dark text-white"
            style={
              Object {
                "bordeRadius": "1rem",
              }
            }
          >
            <div
              className="card-body p-5 text-center"
            >
              <div
                className="mb-md-5 mt-md-4 pb-5"
              >
                <h2
                  className="fw-bold mb-2 text-uppercase"
                >
                  Login
                </h2>
                <p
                  className="text-white-50 mb-5"
                >
                   Enter Your Details
                </p>
                <div
                  className="login_userEmail"
                >
                  <div
                    className="form-outline form-white mb-4"
                  >
                    <input
                      className="form-control form-control-lg"
                      id="typeEmailX"
                      name="userEmail"
                      onChange={[Function]}
                      placeholder="email id"
                      required={true}
                      type="email"
                      value=""
                    />
                    <label
                      className="form-label"
                      for="userEmail"
                    >
                      Email
                    </label>
                  </div>
                </div>
                <div
                  className="login_userPassword"
                >
                  <div
                    className="form-outline form-white mb-4"
                  >
                    <input
                      className="form-control form-control-lg"
                      id="typePasswordX"
                      name="userPassword"
                      onChange={[Function]}
                      placeholder="password"
                      required={true}
                      type="password"
                      value=""
                    />
                    <label
                      className="form-label"
                      for="userPassword"
                    >
                      Password
                    </label>
                  </div>
                </div>
                <p
                  className="small mb-5 pb-lg-2"
                >
                  <a
                    className="text-white-50"
                    href="#!"
                  >
                    Forgot password?
                  </a>
                </p>
                <p
                  defaultValue={false}
                  id="error_paragraph"
                />
                <div
                  className="login_button"
                >
                  <button
                    className="btn btn-outline-light btn-lg px-5"
                    onClick={[Function]}
                    type="submit"
                  >
                    Login
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
